vignettes/miaViz.Rmd

```diff
@@ -326,6 +326,128 @@ names(metadata(dmn_se))
 plotDMNFit(dmn_se, type = "laplace")
 ```
 
+# Serial data ordination and trajectories
+
+```{r, error=FALSE, warning=FALSE, results='hide'}
+if (!requireNamespace("devtools", quietly = TRUE))
+    install.packages("devtools")
+devtools::install_github("microbiome/miaTime")
+```
+
+Principal Coordinate Analysis using Bray-Curtis dissimilarity on the
+`hitchip1006` dataset:
+
+```{r}
+library(dplyr)
+data(hitchip1006, package = "miaTime")
+tse <- hitchip1006
+tse <- transformCounts(tse, method = "relabundance")
+## Ordination with PCoA with Bray-Curtis dissimilarity
+tse <- runMDS(tse, FUN = vegan::vegdist, method = "bray",
+                  name = "PCoA_BC", exprs_values = "relabundance",
+              na.rm = TRUE)
+# Calculating Explained variance
+e <- attr(reducedDim(tse, "PCoA_BC"), "eig")
+rel_eig <- e/sum(e[e>0])
+# plot
+p1 <- plotReducedDim(tse, dimred = "PCoA_BC") +
+    labs(x = paste("PCoA 1 (", round(100 * rel_eig[[1]],1), "%", ")", sep = ""),
+         y = paste("PCoA 2 (", round(100 * rel_eig[[2]],1), "%", ")", sep = ""),
+         title="PCoA - Bray-Curtis")+
+    theme(plot.title = element_text(hjust = 0.5))
+p1
+```
+
+Retrieving information about all available trajectories:
+
+```{r}
+colData(tse)$time <- as.factor(colData(tse)$time)
+table(table(tse$subject)) %>% as.data.frame() %>%
+    rename(Timepoints=Var1, Subjects=Freq)
+```
+
+Lets look at all trajectories having two time points in the data:
+
+```{r}
+# plot
+# List subjects with two time points
+selected.subjects <- names(which(table(tse$subject)==2))
+
+# plot
+p1 <- plotReducedDim(tse, dimred = "PCoA_BC") +
+         labs(title="PCoA - Bray-Curtis")+
+    theme(plot.title = element_text(hjust = 0.5))
+
+# plot
+df <- reducedDim(tse)
+colnames(df) <- paste0("PC", 1:2)
+df <- cbind(df, colData(tse)) %>% as.data.frame()
+
+p1 <- p1 + geom_path(aes(x=PC1, y=PC2, group=subject),
+              arrow=arrow(length = unit(0.1, "inches")),
+              data = subset(df, subject %in% selected.subjects))+ 
+    labs(title = "All trajectories with two time points")
+
+print(p1)
+```
```

Comment:
This can be further simplified by doing something like in this example. Please do.

```
# Load libraries
library(mia)
library(scater)

# Load demo data
data(hitchip1006, package="miaTime")
tse <- hitchip1006

# Run MDS to get reducedDim
tse <- runMDS(tse, exprs_values="counts")

# Plot reducedDim and retrieve ggplot object
p <- plotReducedDim(tse, dimred="MDS")

# Add colData to ggplot object as a separate extra step
p$data <- cbind(p$data, colData(tse))

# Add trajectory between data points from the same subject
p + geom_path(aes(x=X, y=Y, group=subject))
```

Reply:
I personally preferred to have the data passed to rather the argument `data` of `geom_path`, it seems more readable as a code example that not necessarily the same data was used than the one passed originally to the plotReducedDim. This way also further/different subsets or filtering won't affect the p$data and all ordination points are still displayed.

Reply:
Also true and interesting. Ok to do so.

It seems that what is really needed is a way to efficiently pull out a new data.frame that would combine reducedDim and colData, and that could then be used to add new ggplot layers - possibly with subsets - on top of the original plotReducedDim output.

We could replace this:

```
df <- reducedDim(tse)
colnames(df) <- paste0("PC", 1:2)
df <- cbind(df, colData(tse)) %>% as.data.frame()
```

with this:
```
df <- data.frame(reducedDim(tse), colData(tse))
```

in order to simplify code; I think the ordination axis labels do not have to be changed to PC1, PC2 in our example at least, this is something to take care by users, or just change to the final plot?

Reply:
Or even better, just do:

```
p1 <- p1 + geom_path(aes(x=PC1, y=PC2, group=subject),
              arrow=arrow(length = unit(0.1, "inches")),
              data = subset(data.frame(reducedDim(tse), colData(tse)), subject %in% selected.subjects))
```

Reply:
or I can do it also in subsequent examples as the following( without touching the `p` along the way):
`p + geom_path(aes(x=X, y=Y, group=subject),
              data=subset(p$data, subject %in% selected.subjects))`

Reply:
sorry didn't refresh the page to see new comments :)